### PR TITLE
Fix the rule key for S1763 (mistakenly S1736 is used)

### LINKED ISSUE #1973: Fix the rule key for S1763 (mistakenly S1736 is used)
When migrating from legacy rule keys #1925 we by mistake used the wrong number for rule S1763. 
Note that just changing rule key is not enough, we should rely on deprecated rule key logic to allow smooth transition of issues between different rule keys.

### PR DESCRIPTION
**Pull request by andrey-tyukin-sonarsource**
Fixes #1973

### CODE REVIEW

File: javascript-checks/src/main/resources/org/sonar/l10n/javascript/rules/javascript/Sonar_way_recommended_profile.json [hunk 1]
```diff
@@ -52,7 +52,7 @@
     "S1534",
     "S1536",
     "S1656",
-    "S1736",
+    "S1763",
```
Comment (vilchik-elena) [inline javascript-checks/src/main/resources/org/sonar/l10n/javascript/rules/javascript/Sonar_way_recommended_profile.json:55]: move this line to be in order

File: javascript-checks/src/main/resources/org/sonar/l10n/javascript/rules/javascript/Sonar_way_profile.json [hunk 2]
```diff
@@ -31,7 +31,7 @@
     "S1534",
     "S1536",
     "S1656",
-    "S1736",
+    "S1763",
```
Comment (vilchik-elena) [inline javascript-checks/src/main/resources/org/sonar/l10n/javascript/rules/javascript/Sonar_way_profile.json:34]: move this line to be in order

**Comment by andrey-tyukin-sonarsource** [comment]
Please don't merge it just yet. It seems to cause problems on sonarqube side during the validation of transitively deprecated rule keys.

**Comment by vilchik-elena** [comment]
Work on this PR is suspended as there seems to be a bug on SonarQube side not supporting this case

**Comment by saberduck** [comment]
Ticket on SonarQube JIRA https://jira.sonarsource.com/browse/SONAR-13782

**Comment by andrey-tyukin-sonarsource** [comment]
Analogous ticket on SonarCloud JIRA: https://jira.sonarsource.com/browse/SC-2516

**Comment by andrey-tyukin-sonarsource** [comment]
To summarize:

  * The bug has been fixed in SQ: https://jira.sonarsource.com/browse/SONAR-13782 (closed)
  * The bug has been fixed in SC: https://jira.sonarsource.com/browse/SC-2516 (closed)
  * I've re-run the experiment that has been documented [here](https://discuss.sonarsource.com/t/sonarqube-fails-to-start-when-plugins-contain-rules-with-multiple-deprecatedrulekey-s/5144), the error no longer occurs with SonarQube 8.5.1

I therefore assume that the problem is fixed.

**Review by saberduck — APPROVED** [review]

**Comment by roberca2** [comment]
Hello @andrey-tyukin-sonarsource  , @saberduck 

I don't understand exactly how I can fix this issue. I am struggling updating from sonar-javascript-plugin-6.2.1.12157.jar to sonar-javascript-plugin-8.0.0.15689.jar.
I am receiving same error when trying start the container:

2021.06.14 13:51:52 ERROR web[][o.s.s.p.Platform] Background initialization failed. Stopping SonarQube
java.lang.IllegalStateException: An incorrect state of deprecated rule keys has been detected.
The deprecated rule key [javascript:UnreachableCode] was previously deprecated by [javascript:S1736]. [javascript:S1736] should be a deprecated key of [javascript:S1763],

Can you help me fix this? Thanks a lot

**Comment by saberduck** [comment]
Hello @roberca2, upgrade of individual plugins is no longer supported, you need to update whole SonarQube instance. If you face any issue please create post on https://community.sonarsource.com/

**Pull request merged**